Application/EdFi.Ods.Tests/EdFi.Ods.Common/Security/ApiClientAuthenticatorTests.cs

```diff
@@ -8,147 +8,133 @@
             }
 
             protected override void Act()
             {
-                _actualResult = _apiClientAuthenticator.TryAuthenticate("MyInvalidKey", "MySecret", out _apiClientIdentity);
+                _actualResult = _apiClientAuthenticator.TryAuthenticateAsync("MyInvalidKey", "MySecret")
+                    .ConfigureAwait(false)
+                    .GetAwaiter()
+                    .GetResult();;
             }
 
             [Test]
             public void Should_return_false()
             {
-                Assert.That(_actualResult, Is.False);
+                _actualResult.IsAuthenticated.ShouldBeFalse();
             }
 
             [Test]
             public void Should_return_null_identity()
             {
-                Assert.That(_apiClientIdentity, Is.Null);
+                _actualResult.ApiClientDetails.ShouldBeNull();
             }
         }
 
         public class When_trying_to_authenticate_using_valid_key_and_invalid_secret : TestFixtureBase
         {
             private ApiClientIdentity _apiClientIdentity;
-            private bool _actualResult;
+            private ApiClientAuthenticator.AuthenticationResult _actualResult;
             private IApiClientAuthenticator _apiClientAuthenticator;
 
             protected override void Arrange()
             {
-                var apiClientSecret = new ApiClientSecret
-                {
-                    Secret = "MySecret"
-                };
-
-                var apiClientSecretProvider = Stub<IApiClientSecretProvider>();
-
-                A.CallTo(()=> apiClientSecretProvider.GetSecret("MyKey"))
-                    .Returns(apiClientSecret);
-
                 var secretVerifier = Stub<ISecretVerifier>();
 
-                A.CallTo(() => secretVerifier.VerifySecret("MyKey", "MyInvalidSecret", apiClientSecret))
-                    .Returns(false);
+                A.CallTo(() => secretVerifier.VerifySecret("MyKey", "MySecret", 
+                        A<ApiClientSecret>.That.Matches(_ => _.Secret == "MySecret")))
+                    .Returns(true);
 
-                var apiClientIdentityProvider = Stub<IApiClientIdentityProvider>();
+                var apiClientDetailsProvider = Stub<IApiClientDetailsProvider>();
 
-                _apiClientAuthenticator = new ApiClientAuthenticator(apiClientIdentityProvider, apiClientSecretProvider, secretVerifier);
+                _apiClientAuthenticator = new ApiClientAuthenticator(apiClientDetailsProvider, secretVerifier);
             }
 
             protected override void Act()
             {
-                _actualResult = _apiClientAuthenticator.TryAuthenticate("MyKey", "MyInvalidSecret", out _apiClientIdentity);
+                _actualResult = _apiClientAuthenticator.TryAuthenticateAsync("MyKey", "MyInvalidSecret")
+                    .ConfigureAwait(false)
+                    .GetAwaiter()
+                    .GetResult();;
```

Comment:
Remove redundant semi-colon

Reply:
Actually found 3 of them in that file... cleaned up locally and will commit and push all other potential changes once you are done with your review.